### PR TITLE
Don't add `IO#wait*` methods when `RUBY_IO_WAIT_METHODS` is defined by Ruby.

### PR DESCRIPTION
**Pull request by ioquatix**

### CODE REVIEW

File: ext/io/wait/wait.c [hunk 1]
```diff
@@ -177,14 +180,14 @@ io_wait_readable(int argc, VALUE *argv, VALUE io)
 
 #ifndef HAVE_RB_IO_WAIT
     if (wait_for_single_fd(fptr, RB_WAITFD_IN, tv)) {
-	return io;
+        return io;
```
Comment (nobu) [inline ext/io/wait/wait.c:183]: Seems unnecessary.

File: ext/io/wait/wait.c [hunk 2]
```diff
@@ -213,14 +216,14 @@ io_wait_writable(int argc, VALUE *argv, VALUE io)
 #ifndef HAVE_RB_IO_WAIT
     tv = get_timeout(argc, argv, &timerec);
     if (wait_for_single_fd(fptr, RB_WAITFD_OUT, tv)) {
-	return io;
+        return io;
```
Comment (nobu) [inline ext/io/wait/wait.c:219]: ditto

File: ext/io/wait/wait.c [hunk 3]
```diff
@@ -317,61 +333,77 @@ io_wait(int argc, VALUE *argv, VALUE io)
...
     }
     /* rb_time_interval() and might_mode() might convert the argument */
     rb_io_check_closed(fptr);
     if (!event) event = RB_WAITFD_IN;
     if ((event & RB_WAITFD_IN) && rb_io_read_pending(fptr))
-	return Qtrue;
+        return Qtrue;
     if (wait_for_single_fd(fptr, event, tv))
-	return io;
+        return io;
```
Comment (nobu) [inline ext/io/wait/wait.c:349]: ditto.

**Review by nobu — CHANGES_REQUESTED** [review]

File: ext/io/wait/wait.c (hunk 1 above)
Comment (ioquatix) [reply ext/io/wait/wait.c:183]:
```suggestion
	return io;
```

File: ext/io/wait/wait.c (hunk 2 above)
Comment (ioquatix) [reply ext/io/wait/wait.c:219]:
```suggestion
	return io;
```

**Pull request merged**